screens/HomeScreen.js

```diff
@@ -1,35 +1,56 @@
 import React, { useState } from 'react';
-import { Image, Platform, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
-import { Button, Input } from 'react-native-elements';
+import { Button, Platform, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+import { Input } from 'react-native-elements';
 import _sumBy from 'lodash/sumBy';
-import { ScrollView } from 'react-native-gesture-handler';
+
 import * as WebBrowser from 'expo-web-browser';
 
 import { MonoText } from '../components/StyledText';
 import { TextList } from '../components';
 
+const ALL_EXPENSES = [
+  { id: 1, name: 'Buy a book', amount: 20 },
+  { id: 2, name: 'Buy dog food', amount: 5 },
+  { id: 3, name: 'Book a flight ticket', amount: 225 },
+  { id: 4, name: 'Buy speakers', amount: 300 },
+  { id: 5, name: 'Book a concert ticket', amount: 50 },
+  { id: 6, name: 'Buy birthday present', amount: 100 },
+  { id: 7, name: 'Buy boots', amount: 400 },
+  { id: 7, name: 'Buy boots', amount: 1000 },
+];
+
 export default function HomeScreen() {
   const [expenses, setExpenses] = useState(ALL_EXPENSES);
-
-  const ALL_EXPENSES = [
-    { id: 1, name: 'Buy a book', amount: 20 },
-    { id: 2, name: 'Buy dog food', amount: 5 },
-    { id: 3, name: 'Book a flight ticket', amount: 225 },
-    { id: 4, name: 'Buy speakers', amount: 300 },
-    { id: 5, name: 'Book a concert ticket', amount: 50 },
-    { id: 6, name: 'Buy birthday present', amount: 100 },
-    { id: 7, name: 'Buy boots', amount: 400 }
-  ];
-
-  let totalExpense = _sumBy(ALL_EXPENSES, 'amount');
+  const [name, setName] = useState('');
+  const [amount, setAmount] = useState('');
+
+  const handleSubmitForm = () => {
+    if (name !== '' && amount > 0) {
+      const expense = { name, amount };
+      setExpenses([...expenses, expense]);
+      setName('');
+      setAmount('');
+    } else {
+      console.log('Invalid expense name or the amount');
+    }
+  };
+
+  const handleName = (event) => {
+    console.log('Name ', event);
+    setName(event);
+  };
+
+  const handleAmount = (event) => {
+    console.log('Amount', event);
+    setAmount(parseInt(event, 10));
```

Comment:
commenting this method oder line of code would be very nice for saying why this `parseInt` is necessary

Reply:
it's explained  in issue #4 !